tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -216,17 +217,20 @@ public static function sendRequest(
 				$client
 			);
 
-			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY) {
+			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY && $response->getStatusCode() !== self::HTTP_CONFLICT) {
 				$sendExceptionHappened = true;
 			}
 
 			if ($debugResponses) {
 				self::debugResponse($response);
 			}
 			$sendCount = $sendCount + 1;
-			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY) && ($sendCount <= $sendRetryLimit);
+			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY ||
+						($response->getStatusCode() === self::HTTP_CONFLICT &&
+						substr($url, 0, 44) ===self::MAKE_FOLDER_API)) &&
```

Comment:
currently this function has 47 usages but we only need to loop this when we get `409` making a folder so i think we should check for it hear rather than sending a flag from functions to function. or i can change it.